### PR TITLE
strong mode cleanup

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: lib/src/error_group.dart [hunk 1]
```diff
@@ -188,7 +188,7 @@ class _ErrorGroupFuture<T> implements Future<T> {
     _completer.future.catchError((_) {});
   }
 
-  Future/*<S>*/ then/*<S>*/(/*=S*/ onValue(/*=T*/ value), {Function onError}) {
+  Future/*<S>*/ then/*<S>*/(onValue(/*=T*/ value), {Function onError}) {
```
Comment (nex3) [inline lib/src/error_group.dart]: The original declaration was intended. The way it's written now will not be runtime-strong-mode clean.

File: test/cache/repair/reinstalls_git_packages_test.dart [hunk 2]
```diff
@@ -28,7 +28,7 @@ main() {
     pubUpgrade();
 
     // Break them.
-    var fooDirs;
+    List fooDirs;
```
Comment (nex3) [inline test/cache/repair/reinstalls_git_packages_test.dart]: You're welcome to make changes in tests, but be aware that there's no infrastructure in place to make sure they stay strong-mode clean.

File: lib/src/error_group.dart (hunk 1 above)
Comment (kevmoo) [inline lib/src/error_group.dart]: I updated the signature to match Future in the SDK – for which I was given a strong-mode error.

So this is a known delta?

File: test/cache/repair/reinstalls_git_packages_test.dart (hunk 2 above)
Comment (kevmoo) [inline test/cache/repair/reinstalls_git_packages_test.dart]: Noted.

File: lib/src/error_group.dart (hunk 1 above)
Comment (nex3) [inline lib/src/error_group.dart]: Oh I didn't realize this was in a `Future` subclass. That's fine, then.

**Pull request merged**